pkg/errs-dictionary/errdict.go

```diff
@@ -0,0 +1,120 @@
+	LikeyCauseSecondPrefix = "Another possible cause could be "
+)
+
+func fixFormat(s string) string {
+	s = strings.TrimSpace(s)
+	return strings.TrimSuffix(s, ".")
+}
+
+func formatCauses(causes ...string) string {
+	if len(causes) == 0 {
+		return ""
+	}
+	out := LikelyCauseFirstPrefix + fixFormat(causes[0]) + "."
+	for _, c := range causes[1:] {
+		out += LikeyCauseSecondPrefix + fixFormat(c) + "."
+	}
+	return out
+}
+
+// General messages
+const (
+	// Boilerplate messages applicable all over the code base.
+
+	// Action
+	ActionIfErrPersistsContactSupport          = "If this error persists, " + ActionContactSupport
+	ActionIfErrSureCorrectConfigContactSupport = "If you are sure your configuration is correct, " + ActionContactSupport
+	ActionContactSupport                       = "contact support if using managed Istio. Otherwise check if the issue is already fixed in " +
+		"another version of Istio at https://github.com/istio/istio/issues, raise an issue, or ask for help in " +
+		"discuss.istio.io."
+
+	// LikelyCause
+	LikelyCauseAPIServer     = "a problem with the k8s API server"
+	LikelyCauseConfiguration = "an incorrect or badly formatted configuration"
+	LikelyCauseSoftwareBug   = "an issue with the Istio code"
+	LikelyCauseTransience    = "If the error occurred immediately after installation, it is likely permanent."
+
+	// Specific errors, also applicable all over the code base.
+
+	// FailedToGetObjectFromAPIServer
+	ImpactFailedToGetObjectFromAPIServer = "If the error is transient, the impact is low. If permanent, updates for " +
+		"the objects cannot be processed leading to an out of sync control plane."
+	LikelyCauseFailedToGetObjectFromAPIServer = "The most likely cause is a problem with the k8s API server - high " +
+		"load or a transient state."
+)
+
+var (
+	GeneralFailedToGetObjectFromAPIServer = &structured.Error{
+		MoreInfo: "Failed to get an object from the k8s API server, because of a transient " +
+			"error or the object no longer exists.",
+		Impact:      ImpactFailedToGetObjectFromAPIServer,
+		Action:      ActionIfErrPersistsContactSupport,
+		LikelyCause: formatCauses(LikelyCauseFailedToGetObjectFromAPIServer) + " " + LikelyCauseTransience,
+	}
+)
+
+// Operator specific
+const (
+	// NoUpdates
+	OperatorImpactNoUpdates = "In this error state, changes to the IstioOperator CR will not result in the Istio " +
+		"control plane being updated."
+)
+
+var (
+	OperatorFailedToGetObjectInCallback = &structured.Error{
+		MoreInfo: "This error occurred because a k8s update for an IstioOperator CR did not " +
+			"contain an IstioOperator object.",
+		Impact:      OperatorImpactNoUpdates,
+		Action:      ActionIfErrPersistsContactSupport,
+		LikelyCause: formatCauses(LikelyCauseAPIServer) + " " + LikelyCauseTransience,
+	}
+	OperatorFailedToAddFinalizer = &structured.Error{
+		MoreInfo: "This error occurred because a finalizer could not be added to the IstioOperator CR. The " +
```

Comment:
nit: I don't think "CR" should be a user facing term. Resource or object are likely better (not sure what istio.io uses)

Reply:
done.